GoRogue/MapGeneration/PolygonArea.cs

```diff
@@ -130,7 +184,115 @@ private void SetInnerPoints()
+
+
+             Point p1 = origin;
+             Point p2 = origin + new Point(width, 0);
+             Point p3 = origin + new Point(width * 2, height);
+             Point p4 = origin + new Point(width, height);
+
+             return new PolygonArea(algorithm, p1, p2, p3, p4);
+         }
+
+        /// <summary>
+        /// Creates a polygon whose sides are even-length
+        /// </summary>
+        /// <param name="center">The center point of this polygon</param>
+        /// <param name="numberOfSides">Number of sides and corners on this polygon</param>
+        /// <param name="radius">The desired distance between the center and each corner</param>
+        /// <param name="algorithm">Which line-drawing algoirthm to use</param>
+        /// <returns></returns>
+        public static PolygonArea RegularPolygon(Point center, int numberOfSides, double radius, Lines.Algorithm algorithm = Lines.Algorithm.DDA)
+        {
+            if(numberOfSides < 3)
+                throw new ArgumentException("Polygons must have 3 or more sides to be representable in 2 dimensions");
+
+            if (algorithm == Lines.Algorithm.Bresenham || algorithm == Lines.Algorithm.Orthogonal)
+                throw new ArgumentException("Line Algorithm must produce ordered lines.");
+
+            var corners = new List<Point>();
+            var increment = 360.0 / numberOfSides;
+
+            for (int i = 0; i < numberOfSides; i ++)
+            {
+                var theta = SadRogue.Primitives.MathHelpers.ToRadian(i * increment);
+                var corner = new PolarCoordinate(radius, theta).ToCartesian();
+                corner += center;
+                corners.Add(corner);
+            }
+
+            return new PolygonArea(corners, algorithm);
+        }
+
+        /// <summary>
+        /// Creates a new star-shaped polygon
+        /// </summary>
+        /// <param name="center">The center point of the star</param>
+        /// <param name="points">How many points this star has</param>
+        /// <param name="outerRadius">The distance between the center and a tip of the star</param>
+        /// <param name="innerRadius">The distance between the center and an armpit of the star</param>
+        /// <param name="algorithm">Which line-drawing algorithm to use</param>
+        /// <exception cref="ArgumentException">Stars must have 3 or more points; algorithm must be ordered; inner and outer radius must be positive</exception>
+        public static PolygonArea RegularStar(Point center, int points, double outerRadius, double innerRadius,
+            Lines.Algorithm algorithm = Lines.Algorithm.DDA)
+        {
+            if(points < 3)
+                throw new ArgumentException("Polygons must have 3 or more sides to be representable in 2 dimensions");
+            if (algorithm == Lines.Algorithm.Bresenham || algorithm == Lines.Algorithm.Orthogonal)
+                throw new ArgumentException("Line Algorithm must produce ordered lines.");
```

Comment:
Same as above; is there a reason for repeating the constructor checks here?

Reply:
Same as above, I thought it prudent to throw these exceptions before doing any heavy calculations.

Reply:
See above corresponding issue.